src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -156,11 +156,14 @@ internal void ClearCaches()
         internal sealed class CachingContext
         {
             private readonly ConcurrentDictionary<Type, JsonTypeInfo?> _jsonTypeInfoCache = new();
+            private readonly Func<Type, JsonTypeInfo?> _jsonTypeInfoFactory;
```

Comment:
consider adding comment why we're doing this (it's one of the things I'd personally by default choose shorter code)